src/components/TableHead.js

```diff
@@ -19,115 +19,133 @@ const defaultHeadStyles = theme => ({
+        isIndeterminate = true;
+      }
     }
-
-    return (
-      <MuiTableHead
-        className={classNames({
-          [classes.responsiveStacked]:
-            options.responsive === 'vertical' ||
-            options.responsive === 'stacked' ||
-            options.responsive === 'stackedFullWidth',
-          [classes.responsiveSimple]: options.responsive === 'simple',
-          [classes.main]: true,
-        })}>
-        <TableHeadRow>
-          <TableSelectCell
-            ref={el => setCellRef(0, findDOMNode(el))}
-            onChange={this.handleRowSelect.bind(null)}
-            indeterminate={isIndeterminate}
-            checked={isChecked}
-            isHeaderCell={true}
-            expandedRows={expandedRows}
-            expandableRowsHeader={options.expandableRowsHeader}
-            expandableOn={options.expandableRows}
-            selectableOn={options.selectableRows}
-            fixedHeader={options.fixedHeader}
-            fixedSelectColumn={options.fixedSelectColumn}
-            selectableRowsHeader={options.selectableRowsHeader}
-            onExpand={this.props.toggleAllExpandableRows}
-            isRowSelectable={true}
-          />
-          {columns.map(
-            (column, index) =>
-              column.display === 'true' &&
-              (column.customHeadRender ? (
-                column.customHeadRender({ index, ...column }, this.handleToggleColumn, sortOrder)
-              ) : (
-                <TableHeadCell
-                  cellHeaderProps={
-                    columns[index].setCellHeaderProps ? columns[index].setCellHeaderProps({ index, ...column }) : {}
-                  }
-                  key={index}
-                  index={index}
-                  type={'cell'}
-                  ref={el => setCellRef(index + 1, findDOMNode(el))}
-                  sort={column.sort}
-                  sortDirection={column.name === sortOrder.name ? sortOrder.direction : 'none'}
-                  toggleSort={this.handleToggleColumn}
-                  hint={column.hint}
-                  print={column.print}
-                  options={options}
-                  column={column}
-                  components={components}>
-                  {column.label}
-                </TableHeadCell>
-              )),
-          )}
-        </TableHeadRow>
-      </MuiTableHead>
-    );
   }
+
+  let orderedColumns = props.columnOrder.map( colIndex => {
+    return {
+      column: columns[colIndex],
+      index: colIndex,
+    };
+  });
+  
+  return (
+    <MuiTableHead
+      className={classNames({
+        [classes.responsiveStacked]:
+          options.responsive === 'vertical' ||
+          options.responsive === 'stacked' ||
+          options.responsive === 'stackedFullWidth',
+        [classes.responsiveSimple]: options.responsive === 'simple',
+        [classes.main]: true,
+      })}>
+      <TableHeadRow>
+        <TableSelectCell
+          ref={el => setCellRef(0, findDOMNode(el))}
```

Comment:
findDOMNode is oldStyle and heavy

ref : 
- https://reactjs.org/docs/react-dom.html#finddomnode
- https://medium.com/trabe/getting-rid-of-finddomnode-method-in-your-react-application-a0d7093b2660

Reply:
Good catch, I wasn't aware this was on its way out. This code is a hold over from back when Greg originally did the table resize stuff. I'll try and refactor it out as a see it. Looks like there are 8 instances of its use in the code base total.